libshaderc_spvc/src/spvc.cc

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The Shaderc Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "shaderc/spvc.h"
+#include "libshaderc_util/exceptions.h"
+
+#include "spirv-tools/libspirv.hpp"
+#include "spirv-cross/spirv_glsl.hpp"
+#include "spirv-cross/spirv_hlsl.hpp"
+#include "spirv-cross/spirv_msl.hpp"
+
+struct shaderc_spvc_compiler {};
+
+// Described in spvc.h.
+struct shaderc_spvc_compilation_result {
+  std::string validation_messages;
+  std::string compiler_output;
+  shaderc_compilation_status status =
+      shaderc_compilation_status_null_result_object;
```

Comment:
Arguably this could be .....compilation_error?
In the SPIRV-Cross crash case we would get a result object but this default status set. 

Anyway, I'm not to concerned about this.

Reply:
But if it crashed there'd be no one around to check the status?
This is the same in libshaderc.